src/components/annotations/index.js

```diff
@@ -446,8 +446,16 @@ annotations.draw = function(gd, index, opt, value) {
                 // anyway to get its bounding box)
                 if(!ax.autorange && ((options[axLetter] - ax.range[0]) *
                                      (options[axLetter] - ax.range[1]) > 0)) {
-                    annotationIsOffscreen = true;
-                    return;
+                    if(options['a' + axLetter + 'ref'] === axRef) {
+                        if((options['a' + axLetter] - ax.range[0]) *
+                            (options['a' + axLetter] - ax.range[1]) > 0) {
+                            annotationIsOffscreen = true;
+                        }
+                    } else {
+                        annotationIsOffscreen = true;
+                    }
+
+                    if(annotationIsOffscreen) return;
```

Comment:
Hmm. Something isn't quite right still:

![gifrecord_2016-06-17_170451](https://cloud.githubusercontent.com/assets/6675409/16164716/bdea3562-34ad-11e6-8c76-f30a530c6469.gif)

The arrow tip gets pinned to the graph's right edge.